### PR TITLE
Upgrading to Miden v0.6 and adding Merkle Inclusion benchmark

### PR DESCRIPTION
**Pull request by Dominik1999**
I added a new example and upgraded the `benchmarking-cli` to Miden 0.6. 

The new example is basically a Merkle Inclusion Proof for 1000 merkle paths.

### CODE REVIEW

File: benchmarking-cli/src/utils_program.rs
Comment (hackaugusto) [inline benchmarking-cli/src/utils_program.rs:13]: Suggestion: This allows the caller site to be more readable, `MidenProgram::new(prog, DEBUG_OFF);` instead of `MidenProgram::new(prog, false);`. But that is only the case if the user knows about the constant. Instead if you use:

```rust
enum Debug {
  On,
  Off,
}
```

We always get the nicer readability.

File: benchmarking-cli/src/utils_program.rs
Comment (hackaugusto) [inline benchmarking-cli/src/utils_program.rs:24]: It's nice to support both `&str` and `String`. This could be done with:

```suggestion
    pub fn new<S: AsRef<str>>(code_as_str: S, debug: bool) -> Self {
        Self {
            assembler: Assembler::default(),
            masm_code: code_as_str.as_ref().to_string(),
            debug,
            program: None,
            program_info: None,
        }
    }
```

File: benchmarking-cli/src/utils_input.rs
Comment (hackaugusto) [inline benchmarking-cli/src/utils_input.rs:224]: Is this a `TODO`?

File: benchmarking-cli/src/utils_input.rs
Comment (hackaugusto) [inline benchmarking-cli/src/utils_input.rs:228]: Why is there a single `Zero` input? Shouldn't this be just `vec![]`?

File: benchmarking-cli/src/utils_input.rs
Comment (hackaugusto) [inline benchmarking-cli/src/utils_input.rs:199]: A few changes:

- This should handle the case were `word_hex` has an invalid number of bytes (either too many or too little).
- These methods that are not using `self`, are not constructors, and are not coming from a trait, should IMO be standalone functions.

---

Nit: It would also be nice to have this on `crypto` repo, along side this:

https://github.com/0xPolygonMiden/crypto/blob/18302d68e012fb008b5318da47e1525e219b97a8/src/utils/mod.rs#L24-L36

(probably better to open a separate PR and not block this one on that)

File: benchmarking-cli/src/utils_input.rs
Comment (hackaugusto) [inline benchmarking-cli/src/utils_input.rs:62]: Question: We do the same on the `miden` binary, no? Can't we use that code instead of re-implementing it?

**Review by hackaugusto — APPROVED** [review]
thanks! overall looks good, I left some comments to be addressed, once they are done I think we can merge this.

File: examples/merkle_inclusion.masm
Comment (bobbinth) [inline examples/merkle_inclusion.masm:16]: Is there a reason to do both `padw` and then `dropw` inside the loop? I think the following could work just as well:
```
begin
    repeat.1000
        # stack = [i, 0, 0, 0, d, i, R, ...]

        # verify merkle path for node i
        mtree_verify
        # => [i, 0, 0, 0, d, i, R ...]

        add.1 swap.5 add.1
        # => [i+1, 0, 0, 0, d, i+1, R, ...]
    end
end
```

**Review by bobbinth — COMMENTED** [review]
Looks good! Thank you! Just a couple of small comments from my end.

File: benchmarking-cli/src/utils_input.rs
Comment (bobbinth) [reply benchmarking-cli/src/utils_input.rs:62]: I think this is probably copied directly from there. I agree that would be good to use a common component - but this is probably for a different PR. Let's create an issue for this.

File: benchmarking-cli/src/utils_program.rs
Comment (Dominik1999) [reply benchmarking-cli/src/utils_program.rs:24]: this is nice and abstract

File: benchmarking-cli/src/utils_program.rs
Comment (Dominik1999) [reply benchmarking-cli/src/utils_program.rs:13]: I removed it now, I think, no one wants to benchmark in DebugMode

File: benchmarking-cli/src/utils_input.rs
Comment (Dominik1999) [reply benchmarking-cli/src/utils_input.rs:199]: > This should handle the case were word_hex has an invalid number of bytes (either too many or too little).

Will be done in the common file, actually we should use this here 

https://github.com/0xPolygonMiden/miden-vm/blob/5c5749549d5cf1ccb4a6b9c5cf38427a41537145/miden/src/cli/data.rs#L268-L280

**Review by hackaugusto — APPROVED** [review]
LGTM

**Pull request merged**
